File: T4/codigos2122/Recycler/app/src/main/java/com/example/recycler/adapters/AdaptadorRecycler.java
package com.example.recycler.adapters;

import android.content.Context;
import android.graphics.ColorSpace;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.recycler.R;
import com.example.recycler.utils.Modelo;

import java.util.List;

public class AdaptadorRecycler extends RecyclerView.Adapter<AdaptadorRecycler.MyHolder> {


    private List<Modelo> listaElementos;
    private Context context;
    private OnModeloRecyclerListener listener;

    public AdaptadorRecycler(List listaElementos, Context context) {
        this.listaElementos = listaElementos;
        this.context = context;
        this.listener = (OnModeloRecyclerListener) context;

    }

    @NonNull
    @Override
    public MyHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(context).inflate(R.layout.item_recycler,parent,false);
        MyHolder holder = new MyHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyHolder holder, int position) {

        Modelo modeloActual = listaElementos.get(position);
        holder.getImagen().setImageResource(modeloActual.getImagen());
        holder.getTitulo().setText(modeloActual.getModelo());
        holder.getSubTitulo().setText(modeloActual.getMarca());

        holder.getImagen().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //Toast.makeText(context,String.valueOf(modeloActual.getPotencia()),Toast.LENGTH_SHORT).show();
                listener.onModeloSelected(modeloActual);
            }
        });
    }

    @Override
    public int getItemCount() {
        return listaElementos.size();
    }



    class MyHolder extends RecyclerView.ViewHolder {

        private ImageView imagen;
        private TextView titulo, subTitulo;

        public MyHolder(@NonNull View itemView) {
            super(itemView);
            imagen = itemView.findViewById(R.id.imagen_item_recycler);
            titulo = itemView.findViewById(R.id.titulo_item_recycler);
            subTitulo = itemView.findViewById(R.id.subtitulo_item_recycler);
        }

        public ImageView getImagen() {
            return imagen;
        }

        public void setImagen(ImageView imagen) {
            this.imagen = imagen;
        }

        public TextView getTitulo() {
            return titulo;
        }

        public void setTitulo(TextView titulo) {
            this.titulo = titulo;
        }

        public TextView getSubTitulo() {
            return subTitulo;
        }

        public void setSubTitulo(TextView subTitulo) {
            this.subTitulo = subTitulo;
        }
    }

    public interface OnModeloRecyclerListener{

        public void onModeloSelected(Modelo modelo);

    }

}

File: Kotlin/T3/codigos_ces/T3_UI/app/src/main/java/com/example/t3_ui/adapters/AdaptadorRecycler.kt
package com.example.t3_ui.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.core.view.ContentInfoCompat.Flags
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.t3_ui.DetailActivity
import com.example.t3_ui.R
import com.example.t3_ui.model.Modelo
import com.google.android.material.snackbar.Snackbar

class AdaptadorRecycler(
    var lista: ArrayList<Modelo>,
    var contexto: Context
) : RecyclerView.Adapter<AdaptadorRecycler.MyHolder>() {

    private lateinit var listener: OnModeloListener


    init {
        listener = contexto as OnModeloListener
    }

    class MyHolder(item: View) : ViewHolder(item) {
        // elementos
        var imagen: ImageView
        var nombre: TextView
        var precio: TextView
        var botonDetalle: Button
        var botonComparar: Button

        init {
            imagen = item.findViewById(R.id.imagen_modelo)
            nombre = item.findViewById(R.id.nombre_modelo)
            precio = item.findViewById(R.id.precio_modelo)
            botonDetalle = item.findViewById(R.id.boton_detalle)
            botonComparar = item.findViewById(R.id.boton_comparar)

        }
    }

    // indica el numero de los elementos a pintar
    override fun getItemCount(): Int {
        return lista.size
    }

    // crea un patron
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)
            : MyHolder {
        // view -> XML
        val view: View = LayoutInflater.from(contexto)
            .inflate(R.layout.item_recycler, parent, false)
        return MyHolder(view)
    }


    // juntar el patron con los datos
    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        val item: Modelo = lista[position]
        holder.imagen.setImageResource(item.imagen)
        holder.nombre.text = item.nombre
        holder.precio.text = item.precio.toString()
        holder.botonDetalle.setOnClickListener {


            val intent: Intent = Intent(contexto, DetailActivity::class.java);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            intent.putExtra("modelo",item)
            contexto.startActivity(intent)


            /*Snackbar.make(
                holder.boton, "El coche tiene ${item.cv}",
                Snackbar.LENGTH_SHORT
            ).show()*/
        }
        holder.imagen.setOnLongClickListener {
            notifyItemRemoved(position)
            lista.removeAt(position)
            //notifyDataSetChanged()

            return@setOnLongClickListener true
        }
        holder.botonComparar.setOnClickListener {
            listener.onModeloSelected(item)
        }
    }


    interface OnModeloListener{

        fun onModeloSelected(modelo: Modelo)

    }

}